---
date: "2019-05-05T00:00:00+01:00"
title: "Importing data"
linktitle: "1 Importing data"
author: admin
type: book
weight: 1
commentable: true
---



<div id="loading-included-data" class="section level2">
<h2>Loading included data</h2>
<p>You can load the demo data in <code>microbiomedataset</code> package to see the <code>microbiome_dataset</code> class.</p>
<pre class="r"><code>library(microbiomedataset)
library(tidyverse)</code></pre>
<pre class="r"><code>data(&quot;global_patterns&quot;)</code></pre>
<pre class="r"><code>global_patterns
#&gt; -------------------- 
#&gt; microbiomedataset version: 0.99.1 
#&gt; -------------------- 
#&gt; 1.expression_data:[ 19216 x 26 data.frame]
#&gt; 2.sample_info:[ 26 x 8 data.frame]
#&gt; 3.variable_info:[ 19216 x 8 data.frame]
#&gt; 4.sample_info_note:[ 8 x 2 data.frame]
#&gt; 5.variable_info_note:[ 8 x 2 data.frame]
#&gt; -------------------- 
#&gt; Processing information (extract_process_info())
#&gt; create_microbiome_dataset ---------- 
#&gt;             Package               Function.used                Time
#&gt; 1 microbiomedataset create_microbiome_dataset() 2022-07-10 10:56:13</code></pre>
<p>So you can see that we have 1,9216 variables and 26 samples in the dataset.</p>
</div>
<div id="create-microbiome_dataset-class-object" class="section level2">
<h2>Create microbiome_dataset class object</h2>
<p>You can also create the <code>microbiome_dataset</code> class using the <code>create_microbiome_dataset</code> function.</p>
<p>We need to prepare at least three data for it.</p>
<ol style="list-style-type: decimal">
<li><p><code>expression_data</code>: rows are variables and columns are samples.</p></li>
<li><p><code>sample_info</code>: Information for all the samples in <code>expression_data</code>. The first column should be <code>sample_id</code> which should be identical with the column names of <code>expression_data</code>.</p></li>
<li><p><code>variable_info</code>: Information for all the variables in <code>expression_data</code>. The first column should be <code>variable_id</code> which should be identical with the row names of <code>expression_data</code>.</p></li>
</ol>
<pre class="r"><code>expression_data &lt;-
  as.data.frame(matrix(
    sample(1:100, 100, replace = TRUE),
    nrow = 10,
    ncol = 10
  ))

rownames(expression_data) &lt;- paste0(&quot;OTU&quot;, 1:nrow(expression_data))
colnames(expression_data) &lt;-
  paste0(&quot;Sample&quot;, 1:ncol(expression_data))

expression_data
#&gt;       Sample1 Sample2 Sample3 Sample4 Sample5 Sample6 Sample7 Sample8 Sample9
#&gt; OTU1       89      91      32       7      70      79      79      48      52
#&gt; OTU2       40      59      83      58      71      78      17      90      27
#&gt; OTU3       15      71      74      37      58      64      75      66       7
#&gt; OTU4      100      13      49      10      50      88      97      52      70
#&gt; OTU5       81      47      14      17      84      66      35      16      68
#&gt; OTU6       11      95      76      82      59      79      67      88      56
#&gt; OTU7       28       3      88       4      33      78      15      13      64
#&gt; OTU8       67      54      58      69      28      66      99      20      28
#&gt; OTU9       13      91      87      30      94      40      23      18      88
#&gt; OTU10      65      87      71      21       6      46      29      59      50
#&gt;       Sample10
#&gt; OTU1        32
#&gt; OTU2         3
#&gt; OTU3        17
#&gt; OTU4        59
#&gt; OTU5        82
#&gt; OTU6        95
#&gt; OTU7        87
#&gt; OTU8        35
#&gt; OTU9        13
#&gt; OTU10      100

variable_info &lt;-
  as.data.frame(matrix(
    sample(letters, 70, replace = TRUE),
    nrow = nrow(expression_data),
    ncol = 7
  ))

rownames(variable_info) &lt;- rownames(expression_data)
colnames(variable_info) &lt;-
  c(&quot;Domain&quot;,
    &quot;Phylum&quot;,
    &quot;Class&quot;,
    &quot;Order&quot;,
    &quot;Family&quot;,
    &quot;Genus&quot;,
    &quot;Species&quot;)

variable_info$variable_id &lt;-
  rownames(expression_data)

variable_info &lt;-
  variable_info %&gt;% 
  dplyr::select(variable_id, dplyr::everything())

sample_info &lt;-
  data.frame(sample_id = colnames(expression_data),
             class = &quot;Subject&quot;)

object &lt;-
  create_microbiome_dataset(
    expression_data = expression_data,
    sample_info = sample_info,
    variable_info = variable_info
  )

object
#&gt; -------------------- 
#&gt; microbiomedataset version: 0.99.1 
#&gt; -------------------- 
#&gt; 1.expression_data:[ 10 x 10 data.frame]
#&gt; 2.sample_info:[ 10 x 2 data.frame]
#&gt; 3.variable_info:[ 10 x 8 data.frame]
#&gt; 4.sample_info_note:[ 2 x 2 data.frame]
#&gt; 5.variable_info_note:[ 8 x 2 data.frame]
#&gt; -------------------- 
#&gt; Processing information (extract_process_info())
#&gt; create_microbiome_dataset ---------- 
#&gt;             Package               Function.used                Time
#&gt; 1 microbiomedataset create_microbiome_dataset() 2023-09-15 21:07:52</code></pre>
</div>
<div id="convert-phyloseq-class-to-microbiome_dataset-class-object" class="section level2">
<h2>Convert phyloseq class to microbiome_dataset class object</h2>
<p>We can also transfer or convert other common class object to <code>microbiome_dataset</code> class.</p>
<p>Please install <code>phyloseq</code> package at first.</p>
<pre class="r"><code>if(!require(BiocManager)){
  install.packages(&quot;BiocManager&quot;)
}

if(!require(phyloseq)){
  BiocManager::install(&quot;phyloseq&quot;)
}</code></pre>
<pre class="r"><code>library(phyloseq)
data(GlobalPatterns)
GlobalPatterns
#&gt; phyloseq-class experiment-level object
#&gt; otu_table()   OTU Table:         [ 19216 taxa and 26 samples ]
#&gt; sample_data() Sample Data:       [ 26 samples by 7 sample variables ]
#&gt; tax_table()   Taxonomy Table:    [ 19216 taxa by 7 taxonomic ranks ]
#&gt; phy_tree()    Phylogenetic Tree: [ 19216 tips and 19215 internal nodes ]</code></pre>
<p>The first function is <code>convert2microbiome_dataset</code>:</p>
<pre class="r"><code>object1 &lt;- 
  convert2microbiome_dataset(object = GlobalPatterns)
object1
#&gt; -------------------- 
#&gt; microbiomedataset version: 0.99.1 
#&gt; -------------------- 
#&gt; 1.expression_data:[ 19216 x 26 data.frame]
#&gt; 2.sample_info:[ 26 x 8 data.frame]
#&gt; 3.variable_info:[ 19216 x 8 data.frame]
#&gt; 4.sample_info_note:[ 8 x 2 data.frame]
#&gt; 5.variable_info_note:[ 8 x 2 data.frame]
#&gt; -------------------- 
#&gt; Processing information (extract_process_info())
#&gt; create_microbiome_dataset ---------- 
#&gt;             Package               Function.used                Time
#&gt; 1 microbiomedataset create_microbiome_dataset() 2023-09-15 21:08:03</code></pre>
<p>The second function is <code>as.microbiome_dataset</code>:</p>
<pre class="r"><code>object2 &lt;- 
  as.microbiome_dataset(object = GlobalPatterns)
object2
#&gt; -------------------- 
#&gt; microbiomedataset version: 0.99.1 
#&gt; -------------------- 
#&gt; 1.expression_data:[ 19216 x 26 data.frame]
#&gt; 2.sample_info:[ 26 x 8 data.frame]
#&gt; 3.variable_info:[ 19216 x 8 data.frame]
#&gt; 4.sample_info_note:[ 8 x 2 data.frame]
#&gt; 5.variable_info_note:[ 8 x 2 data.frame]
#&gt; -------------------- 
#&gt; Processing information (extract_process_info())
#&gt; create_microbiome_dataset ---------- 
#&gt;             Package               Function.used                Time
#&gt; 1 microbiomedataset create_microbiome_dataset() 2023-09-15 21:08:12</code></pre>
<pre class="r"><code>microbiomedataset::plot_barplot(object = object, 
                                top_n = 5,
                                fill = &quot;Phylum&quot;)</code></pre>
<p><img src="/docs/chapter2/1-import-data_files/figure-html/unnamed-chunk-9-1.png" width="100%" /></p>
<pre class="r"><code>microbiomedataset::plot_barplot(object = object, 
                                top_n = 5,
                                fill = &quot;Phylum&quot;, 
                                relative = TRUE, 
                                re_calculate_relative = TRUE)</code></pre>
<p><img src="/docs/chapter2/1-import-data_files/figure-html/unnamed-chunk-10-1.png" width="100%" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()
#&gt; R version 4.3.0 (2023-04-21)
#&gt; Platform: x86_64-apple-darwin20 (64-bit)
#&gt; Running under: macOS Ventura 13.5.2
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#&gt; 
#&gt; time zone: America/Los_Angeles
#&gt; tzcode source: internal
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] phyloseq_1.44.0           BiocManager_1.30.21      
#&gt;  [3] lubridate_1.9.2           forcats_1.0.0            
#&gt;  [5] stringr_1.5.0             dplyr_1.1.2              
#&gt;  [7] purrr_1.0.1               readr_2.1.4              
#&gt;  [9] tidyr_1.3.0               tibble_3.2.1             
#&gt; [11] ggplot2_3.4.2             tidyverse_2.0.0          
#&gt; [13] microbiomedataset_0.99.10
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.14            
#&gt;   [3] jsonlite_1.8.5              shape_1.4.6                
#&gt;   [5] magrittr_2.0.3              farver_2.1.1               
#&gt;   [7] MALDIquant_1.22.1           rmarkdown_2.22             
#&gt;   [9] GlobalOptions_0.1.2         zlibbioc_1.46.0            
#&gt;  [11] vctrs_0.6.2                 multtest_2.56.0            
#&gt;  [13] RCurl_1.98-1.12             blogdown_1.18.1            
#&gt;  [15] htmltools_0.5.5             S4Arrays_1.0.4             
#&gt;  [17] Rhdf5lib_1.22.0             rhdf5_2.44.0               
#&gt;  [19] gridGraphics_0.5-1          mzID_1.38.0                
#&gt;  [21] sass_0.4.6                  bslib_0.5.0                
#&gt;  [23] htmlwidgets_1.6.2           plyr_1.8.8                 
#&gt;  [25] zoo_1.8-12                  plotly_4.10.2              
#&gt;  [27] impute_1.74.1               cachem_1.0.8               
#&gt;  [29] igraph_1.4.3                lifecycle_1.0.3            
#&gt;  [31] iterators_1.0.14            pkgconfig_2.0.3            
#&gt;  [33] Matrix_1.5-4                R6_2.5.1                   
#&gt;  [35] fastmap_1.1.1               GenomeInfoDbData_1.2.10    
#&gt;  [37] MatrixGenerics_1.12.2       clue_0.3-64                
#&gt;  [39] digest_0.6.31               pcaMethods_1.92.0          
#&gt;  [41] colorspace_2.1-0            masstools_1.0.10           
#&gt;  [43] S4Vectors_0.38.1            rprojroot_2.0.3            
#&gt;  [45] GenomicRanges_1.52.0        vegan_2.6-4                
#&gt;  [47] labeling_0.4.2              timechange_0.2.0           
#&gt;  [49] fansi_1.0.4                 httr_1.4.6                 
#&gt;  [51] mgcv_1.8-42                 polyclip_1.10-4            
#&gt;  [53] compiler_4.3.0              here_1.0.1                 
#&gt;  [55] remotes_2.4.2               withr_2.5.0                
#&gt;  [57] doParallel_1.0.17           BiocParallel_1.34.2        
#&gt;  [59] viridis_0.6.3               highr_0.10                 
#&gt;  [61] ggforce_0.4.1               MASS_7.3-58.4              
#&gt;  [63] DelayedArray_0.26.3         biomformat_1.28.0          
#&gt;  [65] rjson_0.2.21                permute_0.9-7              
#&gt;  [67] ggsci_3.0.0                 mzR_2.34.0                 
#&gt;  [69] tools_4.3.0                 ape_5.7-1                  
#&gt;  [71] zip_2.3.0                   glue_1.6.2                 
#&gt;  [73] rhdf5filters_1.12.1         nlme_3.1-162               
#&gt;  [75] grid_4.3.0                  cluster_2.1.4              
#&gt;  [77] reshape2_1.4.4              ade4_1.7-22                
#&gt;  [79] generics_0.1.3              gtable_0.3.3               
#&gt;  [81] tzdb_0.4.0                  preprocessCore_1.62.1      
#&gt;  [83] data.table_1.14.8           hms_1.1.3                  
#&gt;  [85] tidygraph_1.2.3             utf8_1.2.3                 
#&gt;  [87] XVector_0.40.0              BiocGenerics_0.46.0        
#&gt;  [89] ggrepel_0.9.3               foreach_1.5.2              
#&gt;  [91] pillar_1.9.0                yulab.utils_0.0.6          
#&gt;  [93] limma_3.56.2                splines_4.3.0              
#&gt;  [95] circlize_0.4.15             tweenr_2.0.2               
#&gt;  [97] lattice_0.21-8              survival_3.5-5             
#&gt;  [99] tidyselect_1.2.0            ComplexHeatmap_2.16.0      
#&gt; [101] Biostrings_2.68.1           pbapply_1.7-0              
#&gt; [103] knitr_1.43                  gridExtra_2.3              
#&gt; [105] bookdown_0.34               IRanges_2.34.0             
#&gt; [107] ProtGenerics_1.32.0         SummarizedExperiment_1.30.2
#&gt; [109] Rdisop_1.60.0               stats4_4.3.0               
#&gt; [111] xfun_0.39                   graphlayouts_1.0.0         
#&gt; [113] Biobase_2.60.0              MSnbase_2.26.0             
#&gt; [115] matrixStats_1.0.0           stringi_1.7.12             
#&gt; [117] lazyeval_0.2.2              yaml_2.3.7                 
#&gt; [119] evaluate_0.21               codetools_0.2-19           
#&gt; [121] ggraph_2.1.0                MsCoreUtils_1.12.0         
#&gt; [123] ggplotify_0.1.0             cli_3.6.1                  
#&gt; [125] affyio_1.70.0               munsell_0.5.0              
#&gt; [127] jquerylib_0.1.4             Rcpp_1.0.10                
#&gt; [129] GenomeInfoDb_1.36.0         png_0.1-8                  
#&gt; [131] XML_3.99-0.14               parallel_4.3.0             
#&gt; [133] bitops_1.0-7                tidytree_0.4.2             
#&gt; [135] viridisLite_0.4.2           scales_1.2.1               
#&gt; [137] affy_1.78.0                 openxlsx_4.2.5.2           
#&gt; [139] ncdf4_1.21                  crayon_1.5.2               
#&gt; [141] GetoptLong_1.0.5            rlang_1.1.1                
#&gt; [143] massdataset_1.0.25          vsn_3.68.0</code></pre>
</div>
