---
date: "2019-05-05T00:00:00+01:00"
title: "Merge data"
linktitle: "4 Merge data"
author: admin
type: book
weight: 4
commentable: true
---



<p>Merging OTU or sample indices based on variables in the data can be a useful means of reducing noise or excess features in an analysis or graphic.</p>
<div id="loading-included-data" class="section level2">
<h2>Loading included data</h2>
<pre class="r"><code>library(microbiomedataset)
library(tidyverse)
data(&quot;global_patterns&quot;)
global_patterns
#&gt; -------------------- 
#&gt; microbiomedataset version: 0.99.1 
#&gt; -------------------- 
#&gt; 1.expression_data:[ 19216 x 26 data.frame]
#&gt; 2.sample_info:[ 26 x 8 data.frame]
#&gt; 3.variable_info:[ 19216 x 8 data.frame]
#&gt; 4.sample_info_note:[ 8 x 2 data.frame]
#&gt; 5.variable_info_note:[ 8 x 2 data.frame]
#&gt; -------------------- 
#&gt; Processing information (extract_process_info())
#&gt; create_microbiome_dataset ---------- 
#&gt;             Package               Function.used                Time
#&gt; 1 microbiomedataset create_microbiome_dataset() 2022-07-10 10:56:13</code></pre>
</div>
<div id="merge-samples" class="section level2">
<h2>Merge samples</h2>
<p>Remove empty taxa</p>
<pre class="r"><code>global_patterns2 &lt;-
  global_patterns %&gt;%
  mutate2variable(what = &quot;sum_intensity&quot;) %&gt;%
  activate_microbiome_dataset(what = &quot;variable_info&quot;) %&gt;%
  dplyr::filter(sum_intensity &gt; 0)

humantypes &lt;- c(&quot;Feces&quot;, &quot;Mock&quot;, &quot;Skin&quot;, &quot;Tongue&quot;)
global_patterns2 &lt;-
  global_patterns2 %&gt;%
  activate_microbiome_dataset(what = &quot;sample_info&quot;) %&gt;%
  dplyr::mutate(human = SampleType %in% humantypes)</code></pre>
<p>Now on to the merging examples.</p>
<pre class="r"><code>merged_global_patterns2 &lt;- 
  microbiomedataset::summarise_samples(object = global_patterns2, 
                                       group_by = &quot;SampleType&quot;)
extract_sample_info(merged_global_patterns2)
#&gt;            sample_id  Primer Final_Barcode Barcode_truncated_plus_T
#&gt; 1               Soil ILBC_01        AACGCA                   TGCGTT
#&gt; 2              Feces ILBC_04        AAGAGA                   TCTCTT
#&gt; 3               Skin ILBC_07        AATCGT                   ACGATT
#&gt; 4             Tongue ILBC_10        ACACGA                   TCGTGT
#&gt; 5         Freshwater ILBC_13        ACACTG                   CAGTGT
#&gt; 6 Freshwater (creek) ILBC_16        ACAGCA                   TGCTGT
#&gt; 7              Ocean ILBC_19        ACAGTT                   AACTGT
#&gt; 8 Sediment (estuary) ILBC_22        ACATGT                   ACATGT
#&gt; 9               Mock ILBC_27        ACCGCA                   TGCGGT
#&gt;   Barcode_full_length         SampleType
#&gt; 1         CTAGCGTGCGT               Soil
#&gt; 2         TCGACATCTCT              Feces
#&gt; 3         CGAGTCACGAT               Skin
#&gt; 4         TGTGGCTCGTG             Tongue
#&gt; 5         CATGAACAGTG         Freshwater
#&gt; 6         GACCACTGCTG Freshwater (creek)
#&gt; 7         TCGCGCAACTG              Ocean
#&gt; 8         CACGTGACATG Sediment (estuary)
#&gt; 9         TGACTCTGCGG               Mock
#&gt;                                    Description   class human
#&gt; 1     Calhoun South Carolina Pine soil, pH 4.9 Subject FALSE
#&gt; 2      M3, Day 1, fecal swab, whole body study Subject  TRUE
#&gt; 3      M3, Day 1, right palm, whole body study Subject  TRUE
#&gt; 4         M3, Day 1, tongue, whole body study  Subject  TRUE
#&gt; 5 Lake Mendota Minnesota, 24 meter epilimnion  Subject FALSE
#&gt; 6                 Allequash Creek, 0-1cm depth Subject FALSE
#&gt; 7       Newport Pier, CA surface water, Time 1 Subject FALSE
#&gt; 8               Tijuana River Reserve, depth 1 Subject FALSE
#&gt; 9                                        Even1 Subject  TRUE</code></pre>
</div>
<div id="merge-taxas" class="section level2">
<h2>Merge taxas</h2>
<pre class="r"><code>merged_variables &lt;-
  microbiomedataset::summarize_variables(
    object = global_patterns2,
    variable_index = 1:5,
    remain_variable_info_index = 1
  )
dim(merged_variables)
#&gt; variables   samples 
#&gt;     18984        26
dim(global_patterns2)
#&gt; variables   samples 
#&gt;     18988        26</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()
#&gt; R version 4.3.0 (2023-04-21)
#&gt; Platform: x86_64-apple-darwin20 (64-bit)
#&gt; Running under: macOS Ventura 13.5.2
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#&gt; 
#&gt; time zone: America/Los_Angeles
#&gt; tzcode source: internal
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] lubridate_1.9.2           forcats_1.0.0            
#&gt;  [3] stringr_1.5.0             dplyr_1.1.2              
#&gt;  [5] purrr_1.0.1               readr_2.1.4              
#&gt;  [7] tidyr_1.3.0               tibble_3.2.1             
#&gt;  [9] ggplot2_3.4.2             tidyverse_2.0.0          
#&gt; [11] microbiomedataset_0.99.10
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.14            
#&gt;   [3] jsonlite_1.8.5              shape_1.4.6                
#&gt;   [5] magrittr_2.0.3              farver_2.1.1               
#&gt;   [7] MALDIquant_1.22.1           rmarkdown_2.22             
#&gt;   [9] GlobalOptions_0.1.2         zlibbioc_1.46.0            
#&gt;  [11] vctrs_0.6.2                 multtest_2.56.0            
#&gt;  [13] RCurl_1.98-1.12             blogdown_1.18.1            
#&gt;  [15] htmltools_0.5.5             S4Arrays_1.0.4             
#&gt;  [17] Rhdf5lib_1.22.0             rhdf5_2.44.0               
#&gt;  [19] gridGraphics_0.5-1          mzID_1.38.0                
#&gt;  [21] sass_0.4.6                  bslib_0.5.0                
#&gt;  [23] htmlwidgets_1.6.2           plyr_1.8.8                 
#&gt;  [25] zoo_1.8-12                  plotly_4.10.2              
#&gt;  [27] impute_1.74.1               cachem_1.0.8               
#&gt;  [29] igraph_1.4.3                lifecycle_1.0.3            
#&gt;  [31] iterators_1.0.14            pkgconfig_2.0.3            
#&gt;  [33] Matrix_1.5-4                R6_2.5.1                   
#&gt;  [35] fastmap_1.1.1               GenomeInfoDbData_1.2.10    
#&gt;  [37] MatrixGenerics_1.12.2       clue_0.3-64                
#&gt;  [39] digest_0.6.31               pcaMethods_1.92.0          
#&gt;  [41] colorspace_2.1-0            masstools_1.0.10           
#&gt;  [43] S4Vectors_0.38.1            rprojroot_2.0.3            
#&gt;  [45] GenomicRanges_1.52.0        vegan_2.6-4                
#&gt;  [47] timechange_0.2.0            fansi_1.0.4                
#&gt;  [49] httr_1.4.6                  mgcv_1.8-42                
#&gt;  [51] polyclip_1.10-4             compiler_4.3.0             
#&gt;  [53] here_1.0.1                  remotes_2.4.2              
#&gt;  [55] withr_2.5.0                 doParallel_1.0.17          
#&gt;  [57] BiocParallel_1.34.2         viridis_0.6.3              
#&gt;  [59] ggforce_0.4.1               MASS_7.3-58.4              
#&gt;  [61] DelayedArray_0.26.3         biomformat_1.28.0          
#&gt;  [63] rjson_0.2.21                permute_0.9-7              
#&gt;  [65] ggsci_3.0.0                 mzR_2.34.0                 
#&gt;  [67] tools_4.3.0                 ape_5.7-1                  
#&gt;  [69] zip_2.3.0                   glue_1.6.2                 
#&gt;  [71] rhdf5filters_1.12.1         nlme_3.1-162               
#&gt;  [73] grid_4.3.0                  cluster_2.1.4              
#&gt;  [75] reshape2_1.4.4              ade4_1.7-22                
#&gt;  [77] generics_0.1.3              gtable_0.3.3               
#&gt;  [79] tzdb_0.4.0                  preprocessCore_1.62.1      
#&gt;  [81] data.table_1.14.8           hms_1.1.3                  
#&gt;  [83] tidygraph_1.2.3             utf8_1.2.3                 
#&gt;  [85] XVector_0.40.0              BiocGenerics_0.46.0        
#&gt;  [87] ggrepel_0.9.3               foreach_1.5.2              
#&gt;  [89] pillar_1.9.0                yulab.utils_0.0.6          
#&gt;  [91] limma_3.56.2                splines_4.3.0              
#&gt;  [93] circlize_0.4.15             tweenr_2.0.2               
#&gt;  [95] lattice_0.21-8              survival_3.5-5             
#&gt;  [97] tidyselect_1.2.0            ComplexHeatmap_2.16.0      
#&gt;  [99] Biostrings_2.68.1           pbapply_1.7-0              
#&gt; [101] knitr_1.43                  gridExtra_2.3              
#&gt; [103] bookdown_0.34               phyloseq_1.44.0            
#&gt; [105] IRanges_2.34.0              ProtGenerics_1.32.0        
#&gt; [107] SummarizedExperiment_1.30.2 Rdisop_1.60.0              
#&gt; [109] stats4_4.3.0                xfun_0.39                  
#&gt; [111] graphlayouts_1.0.0          Biobase_2.60.0             
#&gt; [113] MSnbase_2.26.0              matrixStats_1.0.0          
#&gt; [115] stringi_1.7.12              lazyeval_0.2.2             
#&gt; [117] yaml_2.3.7                  evaluate_0.21              
#&gt; [119] codetools_0.2-19            ggraph_2.1.0               
#&gt; [121] MsCoreUtils_1.12.0          BiocManager_1.30.21        
#&gt; [123] ggplotify_0.1.0             cli_3.6.1                  
#&gt; [125] affyio_1.70.0               munsell_0.5.0              
#&gt; [127] jquerylib_0.1.4             Rcpp_1.0.10                
#&gt; [129] GenomeInfoDb_1.36.0         png_0.1-8                  
#&gt; [131] XML_3.99-0.14               parallel_4.3.0             
#&gt; [133] bitops_1.0-7                tidytree_0.4.2             
#&gt; [135] viridisLite_0.4.2           scales_1.2.1               
#&gt; [137] affy_1.78.0                 openxlsx_4.2.5.2           
#&gt; [139] ncdf4_1.21                  crayon_1.5.2               
#&gt; [141] GetoptLong_1.0.5            rlang_1.1.1                
#&gt; [143] massdataset_1.0.25          vsn_3.68.0</code></pre>
</div>
